vllm/worker/model_runner_base.py

```diff
@@ -47,9 +47,14 @@ def _init_attn_metadata_from_tensor_dict(
     valid_attn_kwargs = {}
     for field in dataclasses.fields(attn_backend.get_metadata_cls()):
         val = tensor_dict.pop(field.name, None)
-        if val is not None:
+        # NOTE(kzawora): None is a valid value if type is optional. If
+        # we don't check against it, we will crash by not assigning
+        # Optional types without default value, even if they are
+        # broadcasted properly.
+        is_field_optional = get_origin(field.type) is Union and \
+            type(None) in get_args(field.type)
+        if val is not None or (val is None and is_field_optional):
```

Comment:
Could we not just add detault `None` values to the relevant fields in `HPUAttentionMetadata`? There may be some cost to doing these checks every time.

Alternatively could use some other sentinel value for the `pop` default, assuming `tensor_dict` will contain actual `None` valued entries in these cases?

Or if neither of those are possible for some reason, let's only compute `is_field_optional` in the case that `val` is `None` ....

Reply:
Great idea with the sentinel! Changed now.